browser_library_test/test/node_js_helpers_in_browser_test.dart

```diff
@@ -21,59 +21,69 @@ import 'package:cli_pkg/js.dart';
 import 'package:test/test.dart';
 
 void main() {
-  tearDown(() => delete<Object>(globalThis, 'process'));
+  group('isNodeJs', () {
+    withNonNodeJsProcess(() {
+      test('returns false', () => expect(isNodeJs, isFalse));
+    });
+
+    withFakedNodeJsProcess(() {
+      test('returns true', () => expect(isNodeJs, isTrue));
+    });
+  });
+
+  group('isBrowser', () {
+    withNonNodeJsProcess(() {
+      test('returns true', () => expect(isBrowser, isTrue));
+    });
 
+    withFakedNodeJsProcess(() {
+      test('returns false', () => expect(isBrowser, isFalse));
+    });
+  });
+
+  test('isJS returns true', () => expect(isJS, isTrue));
+
+  // Strict-mode tests are in `npm_test.dart`.
+  test('wrapJSException throws the same primitive in non-strict mode', () {
+    expect(() => wrapJSExceptions(() => throw ''), throwsA(equals('')));
+  });
+}
+
+/// Runs a suite of tests that verify the same behavior across environments that
+/// look like Node.JS, but that don't actually fake a Node.JS environment.
+void withNonNodeJsProcess(void Function() callback) {
   const nonNodeJsProcessTestCases = <String, Map<String, Map<String, String>>>{
     'an empty process': {},
     'a process with empty release': {'release': {}},
     'a process with non-Node.JS release name': {
-      'release': {'name': 'hello'}
+      'release': {'name': 'definitely-not-node'}
     },
   };
 
-  const fakeNodeJsProcess = {
-    'release': {'name': 'node'}
-  };
-
-  group('isNodeJs', () {
-    test("returns 'false' from the browser", () {
-      expect(isNodeJs, isFalse);
-    });
-
-    for (final entry in nonNodeJsProcessTestCases.entries) {
-      final caseName = entry.key;
-      final processJson = entry.value.jsify();
+  group('default environment', callback);
 
-      test("returns 'false' when $caseName exists in the 'window'", () {
-        setProperty(globalThis, 'process', processJson);
-        expect(isNodeJs, isFalse);
-      });
-    }
+  for (final entry in nonNodeJsProcessTestCases.entries) {
+    final caseName = entry.key;
+    final processJson = entry.value.jsify();
 
-    test("returns 'true' with a fake Node.JS process", () {
-      setProperty(globalThis, 'process', fakeNodeJsProcess.jsify());
-      expect(isNodeJs, isTrue);
+    group(caseName, () {
+      tearDown(() => delete<Object>(globalThis, 'process'));
+      setUp(() => setProperty(globalThis, 'process', processJson));
```

Comment:
Write `setUp` before `tearDown` so they appear in the order they execute

Reply:
Ah, too much go for me, I read `teardown()` as `defer` 😅